subsys/net/lib/ptp/clock.c

```diff
@@ -523,13 +523,24 @@ void ptp_clock_synchronize(uint64_t ingress, uint64_t egress)
 	/* If diff is too big, ptp_clk needs to be set first. */
 	if ((offset > (int64_t)NSEC_PER_SEC) || (offset < -(int64_t)NSEC_PER_SEC)) {
 		struct net_ptp_time current;
+		int32_t dest_nsec;
 
 		LOG_WRN("Clock offset exceeds 1 second.");
 
 		ptp_clock_get(ptp_clk.phc, &current);
 
 		current.second -= (uint64_t)(offset / NSEC_PER_SEC);
-		current.nanosecond -= (uint32_t)(offset % NSEC_PER_SEC);
+		dest_nsec = current.nanosecond - (uint32_t)(offset % NSEC_PER_SEC);
```

Comment:
As the variables are different types, it would be good that we add casts here to make it unambiguous what was the intention and to avoid possible Coverity issue

Reply:
@jukkar `nanosecond` field of the `struct net_ptp_time` is `uint32_t` type

Reply:
ohh right, sorry didn't spot that